### DISCUSSION TITLE
hot control

### DISCUSSION THREAD

**Discussion by chrisvidal** [OFFLINE-GmbH/oc-boxes-support#168] [Help] [answered]
My theme is using a `oc-control` that allows,  in that case, to start a skilc() slider when the corresponding `data-control` is on the page.
I am trying to create a box for the editor for this slider.

the `js files` involved is loaded on the editor and also in the frontend but it is not executed.

what am I missing ?

EDIT: the whole html and js and hot control was working with Tailor before switching to Boxes, so I m guessing i am missing something.

**Comment by tobias-kuendig** [comment]
Is it not loading in the editor, the frontend or both? It is hard to tell without seeing code.

 The usage of JS in boxes does not need anything special, except for some init code if you want your widget to survive editor updates in the backend, see https://docs.boxes.offline.ch/use-cases/editor-customization.html#reacting-to-editor-events

The frontend should work without anything else.

**Reply by chrisvidal** [reply]
after cache clearing, it starts to work. I added this in my plugin, pulling the hot control js code directly from the theme

```php
\Event::listen(
            \OFFLINE\Boxes\Classes\Events::EDITOR_RENDER,
            function (\OFFLINE\Boxes\Components\BoxesPageEditor $editor) {

                // $editor->addCss('any-custom.css');
                $editor->addJs('/themes/skillup/assets/js/boxes/hero-slider.js');
            }
        );
```

**Reply by tobias-kuendig** [reply, marked as answer]
Did you see that your box can register assets directly?  There should be no need to add the JS file using the PHP event:

https://docs.boxes.offline.ch/use-cases/adding-assets.html

**Reply by chrisvidal** [reply]
ah I didnt see that possibility indeed. let me try, it looks more logic indeed and easier.
it does replicate the `[resources]` component somehow that I was trying to use.
Does it support `.scss` file as well ?

**Reply by tobias-kuendig** [reply]
It just passes the file along to October's combiner, so it should support SCSS.

https://github.com/OFFLINE-GmbH/oc-boxes-free-plugin/blob/main/classes/traits/HasBoxes.php#L113
